cmd/gaiad/main.go

```diff
@@ -103,16 +101,15 @@ func exportAppStateAndTMValidators(
 	logger log.Logger, db dbm.DB, traceStore io.Writer, height int64, forZeroHeight bool, jailWhiteList []string,
 ) (json.RawMessage, []tmtypes.GenesisValidator, error) {
 
+	var gapp *app.GaiaApp
 	if height != -1 {
-		gapp := app.NewGaiaApp(logger, db, traceStore, false, uint(1), map[int64]bool{}, "")
+		gapp = app.NewGaiaApp(logger, db, traceStore, false, uint(1), map[int64]bool{}, "")
 		err := gapp.LoadHeight(height)
 		if err != nil {
 			return nil, nil, err
 		}
-
-		return gapp.ExportAppStateAndValidators(forZeroHeight, jailWhiteList)
+	} else {
+		gapp = app.NewGaiaApp(logger, db, traceStore, true, uint(1), map[int64]bool{}, "")
 	}
```

Comment:
I'd adopt early return here, i.e.g:

```golang
	if height == -1 {
		return app.NewGaiaApp(logger, db, traceStore, true, uint(1), map[int64]bool{}, "").ExportAppStateAndValidators(forZeroHeight, jailWhiteList)
	}
```

That would allow you to reduce `if-else` to only one `if`. Plus, you reduce indentation and that generally benefits readability.